### PR TITLE
fix: Increase reconciler resources on autopilot

### PR DESCRIPTION
**Pull request by karlkfi**
- Increase requests & limits on autopilot to avoid OOMKill and
  CPU throttling. Leave standard defaults alone, to continue
  using boosting above the minimal requests.
- Clean up createOrPatchDeployment impl
- Use a fake autopilot webhook to test autopilot enablement
- Add TestStressMemoryUsage to test maximum number of supported objects with 100 CRDs and 50 objects each

Depends On:
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/876
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/882
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/910

Proposal: [go/config-sync-reconciler-profiling](http://goto.google.com/config-sync-reconciler-profiling)

b/300294367

### CODE REVIEW

**Comment by karlkfi** [comment]
/retest

unit test flake?

```
    reposync_controller_manager_test.go:266: 
        	Error Trace:	/go/src/kpt.dev/configsync/pkg/reconcilermanager/controllers/reposync_controller_manager_test.go:266
        	Error:      	Received unexpected error:
        	            	Operation cannot be fulfilled on RepoSync.configsync.gke.io "bookinfo/my-repo-sync": ResourceVersion conflict: expected "2" but found "3"
        	Test:       	TestReconcileRepoSyncLifecycleValidToInvalid
```

File: pkg/reconcilermanager/controllers/reconciler_container_resources.go
Comment (sdowell) [inline pkg/reconcilermanager/controllers/reconciler_container_resources.go:76]: Everything LGTM, although I'll let Janet approve these values

/assign @janetkuo

**Review by sdowell — COMMENTED** [review]
Consider cleaning up the commit messages, but LGTM

/lgtm

**Comment by karlkfi** [comment]
/hold

Pulled out math fix to https://github.com/GoogleContainerTools/kpt-config-sync/pull/910

**Comment by karlkfi** [comment]
/unhold

@janetkuo approved the new resources by approving the proposal today: [go/config-sync-reconciler-profiling](http://goto.google.com/config-sync-reconciler-profiling)

**Review by sdowell — APPROVED** [review]

**Pull request merged**
